File: agentcore/src/main/java/agentcore/database/dao/IInputDataDao.java
package agentcore.database.dao;

import agentcore.database.dto.ABaseDtoEntity;

import java.sql.SQLException;

/**
 * Created by user on 21.02.2017.
 */
public interface IInputDataDao<T extends ABaseDtoEntity> {

    public abstract T getFirst() throws SQLException;
    public abstract void delete(T entity) throws SQLException;

}

File: agentcore/src/main/java/agentcore/database/dao/IInputDataDao.kt
package agentcore.database.dao

import agentcore.database.dto.ABaseDtoEntity

import java.sql.SQLException

/**
 * Created by user on 21.02.2017.
 */
interface IInputDataDao<T : ABaseDtoEntity> {

    @Throws(SQLException::class)
    fun getFirst(): T?

    @Throws(SQLException::class)
    fun delete(entity: T)
}
